requirements/small-requirements.txt

```diff
@@ -16,3 +16,4 @@ databricks-cli@git+https://github.com/databricks/databricks-cli.git
 pillow
 matplotlib
 shap>=0.40
+virtualenv
```

Comment:
Should `virtualenv` be included in mlflow's dependencies? I'm leaning towards no because virtualenv support is **still** experimental and optional.

Reply:
or put it in "extra" dependency list ?

Reply:
Including `virtualenv` as an extra dependency sounds good.

Reply:
+1. Placing it in `extras` sgtm.

Reply:
We should also think about how to make it easy for users to install pyenv. Perhaps we should offer a utility through MLflow in MLflow 2.0? We need good docs regardless.

Reply:
Can we just mention https://github.com/pyenv/pyenv#installation somewhere in the doc?

Reply:
Most likely! Just need to make it very obvious that users need to install pyenv.

Reply:
Included the URL in the error message of `_validate_pyenv_is_available`. I'll add a brief instruction containing the link when I update the documentation.